### PR TITLE
Add .rustfmt.toml

### PR DESCRIPTION
**Pull request by peaske7**
Before starting work on https://github.com/ad4mx/spinoff/issues/15, I saw that there was inconsistent formatting applied to the source code, so I added a `.rustfmt.toml` file and ran `cargo fmt`. This PR only contains automatically fixed formatting changes. 

This was to make sure that any future PRs will contain less formatting "noise" and that all contributors get the same output when we run "cargo fmt"

### CODE REVIEW

**Comment by ad4mx** [comment]
Hi, looks good, however is there some way you could set the amount of arguments in a function before separating them vertically in the `.rustfmt.toml`?
Right now, it separates a functions arguments if there are 4 or more, which creates a mess since all the functions have many arguments. If it's possible, could you set it to 5?

**Comment by peaske7** [comment]
rustfmt's defaults are set to maximum character width = 80, so the function declarations (where some exceed 80 characters) will still be realigned vertically. We can set the maximum character width to 100 (from the default 80) if you prefer. 
What do you think?

**Comment by ad4mx** [comment]
Yeah, I think that could work

File: examples/stream.rs [hunk 1]
```diff
@@ -2,7 +2,12 @@ use spinoff::{Spinner, Spinners, Streams};
 use std::{thread::sleep, time::Duration};
 
 fn main() {
-    let sp = Spinner::new_with_stream(Spinners::Aesthetic, "Loading in stderr...", None, Streams::Stderr);
```
Comment (peaske7) [inline examples/stream.rs:5]: For example, this line has `106` characters, so it still gets re-aligned.
I think a max width exceeding 100 is not ideal, so some of the realignments are inescapable. 
What do you think?

Comment (ad4mx) [reply examples/stream.rs:5]: It's fine if it gets re-aligned in files like these, which are just examples. It's just a bit more unreadable when it's in the actual internal code itself.

Comment (peaske7) [reply examples/stream.rs:5]: Yeah, I guess we don't need to do the formatting stuff.

**Pull request closed without merging**